### ISSUE TITLE
OpenAI and Anthropic Pipeline Not working

### ISSUE THREAD

**Issue by MarlNox** [open-webui/pipelines#40]
Both pipelines, in the latest iteration fail to work properly. 
The issues in the OpenAI one, is that setting the key doesn't work using the current structure. 
Trying to interact with the OpenAI pipeline gives a 400 error, related to the api key. 

The OpenAI webui connection works just fine.

**Comment by MarlNox** [comment]
Proposed fix: 

```
from typing import List, Union, Generator, Iterator
from schemas import OpenAIChatMessage
import requests


class Pipeline:
    def __init__(self):
        self.name = "OpenAI Pipeline"
        self.OPENAI_API_KEY = "your-openai-api-key-here"
        self.MODEL = "gpt-3.5-turbo"

    async def on_startup(self):
        print(f"on_startup:{__name__}")
        pass

    async def on_shutdown(self):
        print(f"on_shutdown:{__name__}")
        pass

    def pipe(
        self, user_message: str, model_id: str, messages: List[OpenAIChatMessage], body: dict
    ) -> Union[str, Generator, Iterator]:
        print(f"pipe:{__name__}")

        print(messages)
        print(user_message)

        headers = {
            "Authorization": f"Bearer {self.OPENAI_API_KEY}",
            "Content-Type": "application/json"
        }

        payload = {
            "model": self.MODEL,
            "messages": messages
        }

        if 'stream' in body:
            payload['stream'] = body['stream']

        print(f"Using API Key: {self.OPENAI_API_KEY}")
        print(f"Request Payload: {payload}")

        try:
            r = requests.post(
                url="https://api.openai.com/v1/chat/completions",
                json=payload,
                headers=headers,
                stream=True,
            )

            r.raise_for_status()

            if payload.get("stream"):
                return r.iter_lines()
            else:
                return r.json()
        except requests.exceptions.RequestException as e:
            print(f"Request failed: {e}")
            return f"Error: {e}"


```

**Comment by tjbck** [comment]
Taking a look!

**Comment by tjbck** [comment]
Both should be fixed! Let me know if the issue persists!

**Comment by Lhemamou** [comment]
The issue is persisting for OpenAI (I have only tested this one). I make a PR to fix it #107

**Comment by healdgar** [comment]
Same issue.  It's a 400 error, so I think there's something going on with the URL construction between the openwebUI and the pipeline script for openai.
